monai/utils/deprecated.py

```diff
@@ -44,6 +45,10 @@ def version_leq(lhs, rhs):
     def _try_cast(val):
         val = val.strip()
         try:
+            m = re.match("(\\d+)(.*)", val)
```

Comment:
we should prefer `optional_import("pkg_resources", name="parse_version")` if it is installed... this `version_leq` will be used in many cases, better to make it as robust as possible